maintenance/doMaintenance.php

```diff
@@ -100,6 +100,16 @@
 	require( MWInit::interpretedPath( 'AdminSettings.php' ) );
 }
 $maintenance->finalSetup();
+
+// check if specific task is already running (in case k8s will accidentally create two instances of cron job)
+if ( $maintenance->status->isRunning() ) {
```

Comment:
I don't know what's the risk of failure if we deployed this change, but it clearly breaks backward compatibility.

We completely disallow multiple executions of all scripts. Perhaps we should allow multiple executions by default (backwards compatibility), and explicitly state that some particular script does now allow that (lets us gradually apply that to scripts we are familiar with).

Example: `if ( $maintenance->status->isRunning() && $maintenance->noConcurrency ) { ...`